src/main/scala/mesosphere/mesos/PersistentVolumeMatcher.scala

```diff
@@ -0,0 +1,54 @@
+package mesosphere.mesos
+
+import mesosphere.marathon.core.task.Task
+import mesosphere.marathon.state.AppDefinition
+import org.apache.mesos.{ Protos => Mesos }
+
+import scala.collection.JavaConverters._
+
+object PersistentVolumeMatcher {
+  // FIXME (217): This is completely un-optimized
+  def matchVolumes(
+    offer: Mesos.Offer,
+    app: AppDefinition,
+    waitingTasks: Iterable[Task]): Option[VolumeMatch] = {
+
+    // step 1: find all offered persistent volumes that match this
+    val availableVolumes: Map[String, Mesos.Resource] = offer.getResourcesList.asScala.collect {
+      case resource: Mesos.Resource if resource.hasDisk && resource.getDisk.hasPersistence =>
+        resource.getDisk.getPersistence.getId -> resource
+    }.toMap
+
+    def resourcesForTask(task: Task): Option[Iterable[Mesos.Resource]] = {
+      task.reservationWithVolumes match {
+        // in case there are no required volumes, return a match
+        case None => Some(Nil)
+
+        case Some(reservation) =>
+          val found = reservation.volumeIds.map { id =>
```

Comment:
indeed nicer – I also added comments to these 3 cases and another test case that makes sure the correct task/volume match is returned if there are multiple tasks/volumes